tools/test_api.py

```diff
@@ -2038,7 +2054,12 @@ def find_tests(base_dir, target_name, toolchain_name, app_config=None):
                 if path_depth == 2:
                     test_group_directory_path, test_case_directory = os.path.split(d)
                     test_group_directory = os.path.basename(test_group_directory_path)
-                    
+
+                    # If the target has no network interface configuration, netsocket tests fail to compile
+                    if not module_config and not configs and \
+                        (test_case_directory == 'netsocket' or test_group_directory == 'netsocket'):
+                        continue
```

Comment:
Would that mean the test result is a fail though? Or would they just be skipped?